guides/creating-a-custom-form-field-control.md

```diff
@@ -0,0 +1,337 @@
+It is possible to create custom form field controls that can be used inside `<mat-form-field>`. This
+can be useful if you need to create a components that shares a lot of common behavior with a form
+field, but adds some additional logic.
+
+In order to learn how to build custom form field controls, lets start with a simple input component
+that we want to work inside the form field. For example, a phone number input that segments the
+parts of the number into their own inputs. (Note: this is not intended to be a robust component,
+just a starting point for us to learn.)
+
+<!--
+  TODO(mmalerba): Link to Plunker/StackBlitz with complete example code after beta 12.
+  (Beta 11 is missing some of the features used in the example).
+-->
+
+```ts
+class MyTel {
+  constructor(public area: string, public exchange: string, public subscriber: string) {}
+}
+
+@Component({
+  selector: 'my-tel-input',
+  template: `
+    <style>
+      div {
+        display: flex;
+      }
+      input {
+        border: none;
+        background: none;
+        padding: 0;
+        outline: none;
+        font: inherit;
+        text-align: center;
+      }
+    </style>
```

Comment:
for the sake of cramming it all into 1 markdown box. do you think it would be better to do 2 boxes?

````markdown
```css
  ...
```

```ts
  ...
```
````